prometheus_xmpp/__main__.py

```diff
@@ -158,61 +206,163 @@ def message(self, msg):
         Args:
             msg: The received message stanza.
         """
-        if msg['type'] in ('chat', 'normal'):
-            args = shlex.split(msg['body'])
+        if msg["type"] in ("chat", "normal"):
+            args = shlex.split(msg["body"])
             if args == []:
                 response = "No command specified"
-            elif args[0].lower() in ('alert', 'silence'):
+            elif args[0].lower() in ("alert", "silence"):
                 args[0] = args[0].lower()
-                if msg['from'].bare in self._amtool_allowed:
+                if msg["from"].bare in self._amtool_allowed:
                     if self.alertmanager_url:
-                        args = [
-                            '--alertmanager.url', self.alertmanager_url] + args
+                        args = ["--alertmanager.url", self.alertmanager_url] + args
                     response = run_amtool(args)
                 else:
                     response = "Unauthorized JID."
-            elif args[0].lower() == 'help':
+            elif args[0].lower() == "help":
                 response = "Supported commands: help, alert, silence."
             else:
                 response = "Unknown command: %s" % args[0].lower()
             msg.reply(response).send()
 
+    async def dispatch_message(self, mto, mbody, mhtml):
+        if self.muc == False and self.omemo == False:
+            self.send_message(mto=mto, mbody=mbody, mhtml=mhtml, mtype="chat")
+        elif self.muc == False and self.omemo == True:
+            await self.send_encrypted(mto=[JID(mto)], mtype="chat", body=mbody)
+        elif self.muc == True and self.omemo == False:
+            self.send_message(
+                mto=self.muc_jid, mbody=mbody, mhtml=mhtml, mtype="groupchat"
+            )
+        elif self.muc == True and self.omemo == True:
+            owners, admins, members, outcasts = await asyncio.gather(
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "owner"),
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "admin"),
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "member"),
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "outcast"),
+                return_exceptions=True,
+            )
+            muc_affiliation = owners + admins + members + outcasts
+            recipients = [JID(nick) for nick in muc_affiliation]
+            recipients.remove(self.jid)
+            logging.debug(f"sending encrypted message to recipients: {recipients}")
```

Comment:
Please avoid using f-strings with logging, since that always forces evaluation of the format string even when this particular log level is not used